include/GafferImage/ImageReader.h

```diff
@@ -110,10 +110,18 @@ class ImageReader : public ImageNode
 		Gaffer::StringPlug *colorSpacePlug();
 		const Gaffer::StringPlug *colorSpacePlug() const;
 
+		Gaffer::StringPlug *colorSpaceMetadataNamePlug();
```

Comment:
The thing I liked about the design we discussed is that it enforced some symmetry, so that we could have reasonable confidence that anything we wrote could be read back and would be the same. We had a single mechanism for automatic management of the colourspace (the function) that could be set up at a supervisor level and a single "i know best" mechanism (the colorSpace plug) that could be overridden at an artist level, but this seems to introduce a new mechanism that isn't quite either, and doesn't have equivalent support on the writer.

Maybe a compromise would be to provide the full image metadata to both the reader's and the writer's DefaultColorSpaceFunction, so that if some facilities/shows wanted to use metadata to control colourspace they could do so without putting it under user control?